src/app/shopping-list/shopping-edit/shopping-edit.component.ts

```diff
@@ -1,22 +1,67 @@
-import { Component, OnInit } from '@angular/core';
+import { Subscription } from 'rxjs/Subscription';
+import { Component, OnInit, OnDestroy, ViewChild } from '@angular/core';
 import { ShoppingListService } from '../shopping-list.service';
 import { Ingredient } from 'src/app/shared/model/ingredient.model';
+import { NgForm } from '@angular/forms';
+
 
 
 @Component({
   selector: 'app-shopping-edit',
   templateUrl: './shopping-edit.component.html',
   styleUrls: ['./shopping-edit.component.sass']
 })
-export class ShoppingEditComponent implements OnInit {
+export class ShoppingEditComponent implements OnInit, OnDestroy {
+  @ViewChild('f', null) shoppingListForm: NgForm;
+  startedEditingSubscription: Subscription;
```

Comment:
I do not understand why do you need `startedEditingSubscription`? What do you change that subscription?

Reply:
This variable is needed to be able to unsubscribe.

Reply:
Why do you need this subscription at all? Why don't you pass id through url?

Reply:
I think routing functionality is redundant for this page.

Reply:
I don't think that it's good idea to transfer UI state(edit is UI state) through data service(ShoppingListService). URL + routing is much better for given use case but you can leave as it's